commit db8d93a7a355121d49777c059afbca23c53c8628
Author: Srinivas Dasari <dasaris@codeaurora.org>
Date:   Fri Feb 2 11:15:27 2018 +0200

    nl80211: Fix external_auth check for offloaded authentication
    
    Unfortunately removal of the ext_feature flag in the last revision of
    the patch ended up negating the comparison and prevented the command
    from being processed (either nl80211_external_auth() or
    rdev_external_auth() returns -EOPNOTSUPP). Fix this by adding back the
    lost '!'.
    
    Fixes: 40cbfa90218b ("cfg80211/nl80211: Optional authentication offload to userspace")
    Signed-off-by: Srinivas Dasari <dasaris@codeaurora.org>
    Signed-off-by: Jouni Malinen <jouni@codeaurora.org>
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index 412ed8676306..c6f256b29c73 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -12484,7 +12484,7 @@ static int nl80211_external_auth(struct sk_buff *skb, struct genl_info *info)
 	struct net_device *dev = info->user_ptr[1];
 	struct cfg80211_external_auth_params params;
 
-	if (rdev->ops->external_auth)
+	if (!rdev->ops->external_auth)
 		return -EOPNOTSUPP;
 
 	if (!info->attrs[NL80211_ATTR_SSID])